### PR TITLE
[PM-6328] Checkmarx - Resolve warnings

### PR DESCRIPTION
**Pull request by Hinton**
## Type of change

```
- [ ] Bug fix
- [ ] New feature development
- [X] Tech debt (refactoring, code cleanup, dependency upgrades, etc)
- [ ] Build/deploy pipeline (DevOps)
- [ ] Other
```

## Objective

Resolve Checkmarx warnings and add some comments explaining some invalid Checkmarx warnings.

`target="_blank"` implies `noopener`, but we still want to set `rel="noreferrer"` to remove the referring header.

## Before you submit

- Please add **unit tests** where it makes sense to do so (encouraged but not required)
- If this change requires a **documentation update** - notify the documentation team
- If this change has particular **deployment requirements** - notify the DevOps team
- Ensure that all UI additions follow [WCAG AA requirements](https://contributing.bitwarden.com/contributing/accessibility/)

### CODE REVIEW

File: apps/browser/src/auth/popup/register.component.html [hunk 1]
```diff
@@ -132,11 +132,11 @@ <h1 class="center">
           <input type="checkbox" id="acceptPolicies" formControlName="acceptPolicies" />
           <label for="acceptPolicies">
             {{ "acceptPolicies" | i18n }}<br />
-            <a href="https://bitwarden.com/terms/" target="_blank" rel="noopener">{{
+            <a href="https://bitwarden.com/terms/" target="_blank" rel="noopener noreferrer">{{
```
Comment (withinfocus) [inline apps/browser/src/auth/popup/register.component.html:135]: ⚠️ We can also remove "noopener" when we do this.

**Review by withinfocus — APPROVED** [review]
Clean and easy changes.

**Review by shane-melton — APPROVED** [review]

**Review by djsmith85 — APPROVED** [review]

**Review by JaredSnider-Bitwarden — APPROVED** [review]

**Review by cagonzalezcs — APPROVED** [review]

**Review by vincentsalucci — APPROVED** [review]
AC files approved

**Review by Thomas-Avery — APPROVED** [review]
SM changes approved

**Review by addisonbeck — APPROVED** [review]

**Review by MGibson1 — APPROVED** [review]

**Review by willmartian — APPROVED** [review]

**Pull request merged**